src/unreadCards/UnreadCard.js

```diff
@@ -0,0 +1,186 @@
+    isPrivate: boolean,
+    sender?: string,
+    onSwipe: () => void,
+    onPress: () => void,
+  };
+
+  animatedPosition = new Animated.ValueXY();
+  animatedHeight = new Animated.Value(1);
+
+  panResponder = PanResponder.create({
+    onStartShouldSetPanResponder: () => true,
+    onStartShouldSetPanResponderCapture: () => false,
+    onMoveShouldSetPanResponder: () => true,
+    onPanResponderMove: (event, gesture) => {
+      const xOffset = gesture.dx;
+      if (xOffset > 0) {
+        this.animatedPosition.setValue({ x: xOffset, y: gesture.dy });
+      }
+    },
+    onPanResponderRelease: (event, gesture) => {
+      this.handleTouchEnd(gesture.dx);
+    },
+    onPanResponderReject: (e, gesture) => {
+      this.handleTouchEnd(gesture.dx);
+    },
+    onPanResponderTerminate: (e, gesture) => {
+      this.handleTouchEnd(gesture.dx);
+    },
+    onPanResponderEnd: (e, gesture) => {
+      this.handleTouchEnd(gesture.dx);
+    },
+  });
+
+  handleTouchEnd = (xOffset: number) => {
+    if (xOffset > SWIPE_THRESHOLD) {
+      this.forceSwipe();
+    } else {
+      this.resetPosition();
+    }
+  };
+
+  onSwipeComplete = () => {
+    const { onSwipe } = this.props;
+
+    Animated.timing(this.animatedHeight, {
+      toValue: 0,
+      duration: SWIPE_OUT_DURATION,
+    }).start(onSwipe);
+  };
+
+  onPress = () => {
+    this.props.onPress();
+  };
+
+  // If touch ends beyond SWIPE_THRESHOLD swipe the card out forcefully.
+  forceSwipe = () => {
+    Animated.timing(this.animatedPosition, {
+      toValue: { x: SCREEN_WIDTH, y: 0 },
+      duration: SWIPE_OUT_DURATION,
+    }).start(() => this.onSwipeComplete());
+  };
+
+  // If touch ends before SWIPE_THRESHOLD restore the card's position.
+  resetPosition = () => {
+    Animated.spring(this.animatedPosition, {
+      toValue: { x: 0, y: 0 },
+    }).start();
+  };
+
+  dynamicContainerStyles = () => [
+    styles.container,
+    {
+      height: this.animatedHeight.interpolate({
+        inputRange: [0, 1],
+        outputRange: [0, this.state.cardHeight],
+      }),
+    },
+  ];
+
+  dynamicReadIndicatorStyles = () => ({
+    position: 'absolute',
+    left: 10,
+    flexDirection: 'column',
+    opacity: this.animatedPosition.x.interpolate({
+      inputRange: [SWIPE_THRESHOLD, SCREEN_WIDTH * 0.9],
+      outputRange: [1, 0],
+    }),
+    transform: [
+      {
+        translateX: this.animatedPosition.x.interpolate({
+          inputRange: [0, SWIPE_THRESHOLD],
+          outputRange: [0, 20],
+        }),
+      },
+    ],
+  });
+
+  // Called when component is first laid out to dynamically calculate the
+  // height of the card and make the card container of the same height.
+  measureView = event => {
+    this.setState({
+      cardHeight: event.nativeEvent.layout.height,
+    });
+  };
+
+  render() {
+    const { stream, color, topics, unreadCount, isPrivate, sender } = this.props;
+
+    return (
+      <Animated.View style={this.dynamicContainerStyles()}>
```

Comment:
You can separate the animated functionality from the content functionality.
Have a look at `AnimatedScaleComponent` and `AnimatedHeightComponent` for ideas.

Reply:
Did you see https://github.com/jshanson7/react-native-swipeable ?

Reply:
Actually, it seems React Native already has support for swipeable stuff. They just did not document it:
https://github.com/facebook/react-native/tree/master/Libraries/Experimental/SwipeableRow